pkg/commands/entrypoints/entrypoints.go

```diff
@@ -0,0 +1,251 @@
+	flagSkipDetectChanges       bool
+	flagFormat                  string
+	flagFailUnresolvableModules bool
+
+	gitClient git.Git
+}
+
+func (c *EntrypointsCommand) Desc() string {
+	return `Determine the entrypoint directories to run Guardian commands`
+}
+
+func (c *EntrypointsCommand) Help() string {
+	return `
+Usage: {{ COMMAND }} [options] <directory>
+
+	Determine the entrypoint directories to run Guardian commands.
+`
+}
+
+func (c *EntrypointsCommand) Flags() *cli.FlagSet {
+	set := c.NewFlagSet()
+
+	c.GitHubFlags.Register(set)
+	c.RetryFlags.Register(set)
+
+	f := set.NewSection("COMMAND OPTIONS")
+
+	f.IntVar(&cli.IntVar{
+		Name:    "pull-request-number",
+		Target:  &c.flagPullRequestNumber,
+		Example: "100",
+		Usage:   "The GitHub pull request number associated with this plan run.",
+	})
+
+	f.StringVar(&cli.StringVar{
+		Name:    "dest-ref",
+		Target:  &c.flagDestRef,
+		Example: "ref-name",
+		Usage:   "The destination GitHub ref name for finding file changes.",
+	})
+
+	f.StringVar(&cli.StringVar{
+		Name:    "source-ref",
+		Target:  &c.flagSourceRef,
+		Example: "ref-name",
+		Usage:   "The source GitHub ref name for finding file changes.",
+	})
+
+	f.BoolVar(&cli.BoolVar{
+		Name:   "skip-detect-changes",
+		Target: &c.flagSkipDetectChanges,
+		Usage:  "Skip detecting file changes and run for all entrypoint directories.",
+	})
+
+	f.StringVar(&cli.StringVar{
+		Name:   "format",
+		Target: &c.flagFormat,
+		Usage:  fmt.Sprintf("The format to print the output directories. The supported formats are: %s.", strings.Join(maps.Keys(allowedFormats), ", ")),
+	})
+
+	f.BoolVar(&cli.BoolVar{
+		Name:    "fail-unresolvable-modules",
+		Target:  &c.flagFailUnresolvableModules,
+		Usage:   `Whether or not to error if a module cannot be resolved.`,
+		Default: false,
+	})
+
+	return set
+}
+
+func (c *EntrypointsCommand) Run(ctx context.Context, args []string) error {
+	f := c.Flags()
+	if err := f.Parse(args); err != nil {
+		return fmt.Errorf("failed to parse flags: %w", err)
+	}
+
+	parsedArgs := f.Args()
+
+	if len(parsedArgs) != 1 {
+		return flag.ErrHelp
+	}
+
+	dirAbs, err := util.PathEvalAbs(parsedArgs[0])
+	if err != nil {
+		return fmt.Errorf("failed to absolute path for directory: %w", err)
+	}
+	c.directory = dirAbs
+
+	c.gitClient = git.NewGitClient(c.directory)
+
+	return c.Process(ctx)
+}
+
+// Process handles the main logic for the Guardian init process.
+func (c *EntrypointsCommand) Process(ctx context.Context) error {
+	logger := logging.FromContext(ctx).
+		With("github_owner", c.GitHubFlags.FlagGitHubOwner).
+		With("github_repo", c.GitHubFlags.FlagGitHubOwner).
+		With("pull_request_number", c.flagPullRequestNumber)
+
+	logger.DebugContext(ctx, "Starting Guardian init")
+
+	if c.flagFormat == "" {
+		c.flagFormat = "text"
+	}
+
+	if _, ok := allowedFormats[c.flagFormat]; !ok {
+		return fmt.Errorf("invalid format flag: %s (supported formats are: %s)", c.flagFormat, strings.Join(maps.Keys(allowedFormats), ", "))
+	}
```

Comment:
It would be better to do this kind of validation/defaulting in [AfterParse](https://github.com/abcxyz/pkg/blob/9b36be5112d6211f33d81a8ebcb29693e66a3809/cli/command_doc_after_parse_test.go#L66-L78). Then you can easily test it too.

Reply:
Ah, nice!